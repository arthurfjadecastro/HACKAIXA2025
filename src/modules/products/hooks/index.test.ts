import * as hooksIndex from './index';

describe('Products Hooks Index', () => {
  it('should export all product hooks', () => {
    expect(hooksIndex).toBeDefined();
    expect(typeof hooksIndex).toBe('object');
  });

  it('should export useCreateProduct hook', () => {
    expect(hooksIndex.useCreateProduct).toBeDefined();
    expect(typeof hooksIndex.useCreateProduct).toBe('function');
  });

  it('should export useDeleteProduct hook', () => {
    expect(hooksIndex.useDeleteProduct).toBeDefined();
    expect(typeof hooksIndex.useDeleteProduct).toBe('function');
  });

  it('should export useProducts hook', () => {
    expect(hooksIndex.useProducts).toBeDefined();
    expect(typeof hooksIndex.useProducts).toBe('function');
  });

  it('should export useProductManagement hook', () => {
    expect(hooksIndex.useProductManagement).toBeDefined();
    expect(typeof hooksIndex.useProductManagement).toBe('function');
  });

  it('should match snapshot', () => {
    expect(Object.keys(hooksIndex).sort()).toMatchSnapshot();
  });
});
