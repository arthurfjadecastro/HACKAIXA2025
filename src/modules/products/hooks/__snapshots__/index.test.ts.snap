// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Hooks Index should match snapshot 1`] = `
[
  "useCreateProduct",
  "useDeleteProduct",
  "useProductManagement",
  "useProducts",
]
`;
